addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -757,7 +757,7 @@ export class SelfOrder extends Reactive {
     }
 
     formatMonetary(price) {
-        return webFormatCurrency(price, this.currency_id);
+        return webFormatCurrency(price, this.currency?.id);
```

Comment:
Why do you put a question mark ? I think is impossible to not have the currency ?

Reply:
![image](https://github.com/user-attachments/assets/87114977-ec8b-4985-a526-2c95007527cb)
Hello @davidmonnom 

I am getting the above error in test cases.

Thanks

Reply:
Hello @davidmonnom 

The error mentioned above occurred because the `formatMonetary` function of `self_order_service` was called during receipt printing using the OrderReceipt component. As a result, `this.currency` was undefined, which caused an error when printing the receipt. To resolve this issue, we need to pass the `currency_id` in the data while printing the receipt and use that `currency_id` in the `formatMonetary` function of `self_order_service`.

Thanks

Reply:
Can you try `formatCurrency: this.formatMonetary.bind(this),` ?